ledger/participant-integration-api/src/bench/platform/store/interning/BenchmarkState.scala

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.interning
+
+import org.openjdk.jmh.annotations.{Level, Param, Scope, Setup, State}
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.Random
+
+@State(Scope.Benchmark)
+abstract class BenchmarkState {
+  @Param(Array("10000", "100000", "1000000", "10000000"))
+  var stringCount: Int = _
+
+  @Param(Array("10", "100"))
+  var stringLength: Int = _
+
+  protected val perfTestTimeout: FiniteDuration = 5.minutes
+
+  protected var entries: Array[(Int, String)] = _
+  protected var interning: StringInterningView = _
+  protected var interningEnd: Int = _
+
+  protected def extraStringCount = 0
+
+  @Setup(Level.Trial)
+  def setupEntries(): Unit = {
+    entries = BenchmarkState.createEntries(stringCount + extraStringCount, stringLength)
+  }
+}
+
+object BenchmarkState {
+
+  protected val perfTestTimeout: FiniteDuration = 5.minutes
+
+  private[this] def randomString(length: Int): String = Random.alphanumeric.take(length).mkString
+
+  def createEntries(stringCount: Int, stringLength: Int): Array[(Int, String)] = {
+    Console.print(
+      s"Creating an array with $stringCount entries with string length $stringLength..."
+    )
+
+    val entries = new Array[(Int, String)](stringCount)
+    (0 until stringCount).foreach(i => entries(i) = (i + 1) -> randomString(stringLength))
```

Comment:
We are using Map as the vehicle behind the interning. I fear that perfectly random strings will behave better in that regard than real-life strings with common prefixes.

Reply:
We intern party and template names. Template names are already random, party names are random at least in canton.

Also, Map is implemented by HashMap, so the distribution of strings should not matter that much.